share/cmake/modules/FindHalf.cmake

```diff
@@ -0,0 +1,226 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+#
+# Locate or install ilmbase
+#
+# Variables defined by this module:
+#   Half_FOUND - If FALSE, do not try to link to ilmbase
+#   Half_LIBRARY - Half library to link to
+#   Half_INCLUDE_DIR - Where to find half.h
+#   Half_VERSION - The version of the library
```

Comment:
Based on the global comment for the pull request, I understand that each third-party library also supports _ROOT variable such as half_ROOT.

Reply:
`_ROOT` is mentioned in the paragraph below this list of variables. It is not included in this list because it is not set by the find module. This is the note I added for each module:

```
# If ilmbase is not installed in a standard path, you can use the 
# Half_ROOT variable to tell CMake where to find it. If it is not found 
# and OCIO_INSTALL_EXT_PACKAGES is set to MISSING or ALL, ilmbase will be 
# downloaded, built, and statically-linked into libOpenColorIO at build time.
```

Reply:
Though seeing now that I should change `ilmbase` to `Half` in that message for clarity